pyrokinetics/equilibrium.py

```diff
@@ -159,26 +159,40 @@ def read_geqdsk(self, psi_n_lcfs=1.0):
     def get_flux_surface(self, psi_n):
 
         import matplotlib as mpl
-
-        mpl.use("Agg")
         import matplotlib.pyplot as plt
 
         # Generate 2D mesh of normalised psi
         psi_2d = np.transpose(self.psi_RZ(self.R, self.Z))
 
         psin_2d = (psi_2d - self.psi_axis) / (self.psi_bdry - self.psi_axis)
 
-        # Returns a list of list of contours for psi_n
+        # Returns a list of list of contours for psi_n, resets backend to original value
+        original_backend = mpl.get_backend()
+        mpl.use("Agg")
+
         con = plt.contour(self.R, self.Z, psin_2d, levels=[0, psi_n])
 
+        mpl.use(original_backend)
+
         paths = con.collections[1].get_paths()
 
         if psi_n == 1.0 and len(paths) == 0:
             raise ValueError(
                 "PsiN=1.0 for LCFS isn't well defined. Try lowering psi_n_lcfs"
             )
 
-        path = paths[np.argmax(len(paths))]
+        # Find smallest path integral to find closed loop
+        loop_integrals = []
+        for path in paths:
```

Comment:
Looks good, I do something similar in my step_utils repo but instead look for the path with the lowest distance from rmaxis, zmaxis. If the equilibrium output contains coils on the grid (e.g. a free boundary code) then there can be closed loops around them.

Reply:
I did think about that but I wasn't sure if there could be cases where this might not be true. I guess if we compare the the closest point of each contour to raxis and zaxis, then the smallest one must be the closed loop?
 
Worth changing?

Reply:
My code for this is

~~~~
        if len(results) == 1:
            result = results[0]
        else:
            # Otherwise find the path with the least average distance from rmaxis, zmaxis
            distances = [ np.mean( np.sqrt((x.vertices[:,0] - self['rmaxis'])**2 +
                                           (x.vertices[:,1] - self['zmaxis'])**2)) for x in results]
            result = results[np.array(distances).argmin()]
~~~~

where results if your paths.

Probably not very likely to hit currently but it could be an issue with free boundary sources.

Reply:
@d7919  Just updated to this method